packages/swapper/src/swappers/ArbitrumBridgeSwapper/utils/fetchArbitrumBridgeSwap.ts

```diff
@@ -67,44 +75,98 @@ export const fetchArbitrumBridgeSwap = async ({
+          const { gasPrice, maxFeePerGas } = average
+
+          // eip1559 fees
+          if (supportsEIP1559 && maxFeePerGas) {
+            return fallbackEthDepositGasLimit.times(maxFeePerGas).toFixed()
+          }
+
+          // legacy fees
+          return fallbackEthDepositGasLimit.times(gasPrice).toFixed()
+        }
+
+        // Actual fees
+        const feeData = await evm.getFees({
+          adapter,
+          data: maybeRequest.txRequest.data.toString(),
+          to: maybeRequest.txRequest.to,
+          value: maybeRequest.txRequest.value.toString(),
+          from: maybeRequest.txRequest.from,
+          supportsEIP1559,
+        })
+
+        return feeData.networkFeeCryptoBaseUnit
+      })()
+
+      return { request: maybeRequest, networkFeeCryptoBaseUnit, allowanceContract: '0x0' }
     }
     case BRIDGE_TYPE.ETH_WITHDRAWAL: {
       const bridger = new EthBridger(l2Network)
 
-      const request = await bridger.getWithdrawalRequest({
-        amount: BigNumber.from(sellAmountIncludingProtocolFeesCryptoBaseUnit),
-        from: sendAddress ?? '',
-        destinationAddress: receiveAddress,
-      })
-
-      const { networkFeeCryptoBaseUnit } = await evm.getFees({
-        adapter,
-        data: request.txRequest.data.toString(),
-        to: request.txRequest.to,
-        value: request.txRequest.value.toString(),
-        from: request.txRequest.from,
-        supportsEIP1559,
-      })
-
-      return { request, allowanceContract: '0x0', networkFeeCryptoBaseUnit }
+      const maybeRequest =
+        priceOrQuote === 'quote'
+          ? await bridger
+              .getWithdrawalRequest({
+                amount: BigNumber.from(sellAmountIncludingProtocolFeesCryptoBaseUnit),
+                from: sendAddress!,
+                destinationAddress: receiveAddress!,
+              })
+              .catch(e => {
+                console.error('Error getting ETH withdraw request', e)
+                return undefined
+              })
+          : undefined
+
+      const networkFeeCryptoBaseUnit = await (async () => {
+        // Fallback fees
+        if (!maybeRequest) {
+          const { average } = await adapter.getGasFeeData()
+          const { gasPrice, maxFeePerGas } = average
+
+          // eip1559 fees
+          if (supportsEIP1559 && maxFeePerGas) {
+            return fallbackEthWithdrawGasLimit.times(maxFeePerGas).toFixed()
+          }
+
+          // legacy fees
+          return fallbackEthWithdrawGasLimit.times(gasPrice).toFixed()
+        }
+
+        // Actual fees
+        const feeData = await evm.getFees({
+          adapter,
+          data: maybeRequest.txRequest.data.toString(),
+          to: maybeRequest.txRequest.to,
+          value: maybeRequest.txRequest.value.toString(),
+          from: maybeRequest.txRequest.from,
+          supportsEIP1559,
+        })
+
+        return feeData.networkFeeCryptoBaseUnit
+      })()
```

Comment:
Nitpick: here too - can we lift this whole IIFE into a util?

Reply:
Added to https://github.com/shapeshift/web/pull/7986 TODOs

Reply:
gm https://github.com/shapeshift/web/pull/7986/commits/4b1a8141d8b34077f2e9e24828751fccd14f0f75